### PR TITLE
[WS2_32_APITEST:WSAASYNC] Improve this test. ROSTESTS-317

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

JIRA issue: [ROSTESTS-317](https://jira.reactos.org/browse/ROSTESTS-317)

Result examples, before clean fix:
[(WTB) Job 45070](https://testbot.winehq.org/JobDetails.pl?Key=45070) // Gone :-(
[(WTB) Job 45076](https://testbot.winehq.org/JobDetails.pl?Key=45076) // Gone :-(

### CODE REVIEW

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 1]
```diff
@@ -13,11 +13,12 @@
...
-    SOCKET     ServerSocket = INVALID_SOCKET,
-               ClientSocket = INVALID_SOCKET;
-    WSAEVENT   ServerEvent = WSA_INVALID_EVENT,
-               ClientEvent = WSA_INVALID_EVENT;
+    int err;
+    WSADATA WsaData;
+    SOCKET ServerSocket = INVALID_SOCKET,
+           ClientSocket = INVALID_SOCKET;
+    WSAEVENT ServerEvent = WSA_INVALID_EVENT,
+             ClientEvent = WSA_INVALID_EVENT;
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: why?

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 2]
```diff
@@ -26,78 +27,77 @@ START_TEST(WSAAsync)
...
     if (ent == NULL)
     {
-        ok(ent != NULL, "ERROR: gethostbyname '127.0.0.1' failed, trying 'localhost'\n");
         ent = gethostbyname("localhost");
+        ok(ent != NULL, "gethostbyname(\"localhost\") failed too: %d\n", WSAGetLastError());
 
         if (ent == NULL)
         {
-            skip("ERROR: gethostbyname 'localhost' failed\n");
+            skip("No host\n");
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why?

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 3]
```diff
@@ -26,78 +27,77 @@ START_TEST(WSAAsync)
...
+    if (ServerSocket == INVALID_SOCKET ||
+        ClientSocket == INVALID_SOCKET)
+    {
+        skip("No ServerSocket/ClientSocket\n");
+        goto done;
+    }
+
     server_addr_in.sin_family = AF_INET;
-    server_addr_in.sin_port   = htons(SVR_PORT);
+    server_addr_in.sin_port = htons(SVR_PORT);
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 4]
```diff
@@ -26,78 +27,77 @@ START_TEST(WSAAsync)
...
     memcpy(&server_addr_in.sin_addr.S_un.S_addr, ent->h_addr_list[0], 4);
 
-    // server inialialization
+    // Server initialization.
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 5]
```diff
@@ -26,78 +27,77 @@ START_TEST(WSAAsync)
...
     trace("Initializing server and client connections ...\n");
     ok(bind(ServerSocket, (struct sockaddr*)&server_addr_in, sizeof(server_addr_in)) == 0, "ERROR: server bind failed\n");
     ok(ioctlsocket(ServerSocket, FIONBIO, &ulValue) == 0, "ERROR: server ioctlsocket FIONBIO failed\n");
     ok(WSAEventSelect(ServerSocket, ServerEvent, FD_ACCEPT | FD_CLOSE) == 0, "ERROR: server accept EventSelect failed\n");
 
-    // client inialialization
+    // Client initialization.
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 6]
```diff
@@ -157,48 +158,48 @@ START_TEST(WSAAsync)
...
+        skip("No ServerSocket/ClientSocket\n");
+        goto done;
     }
 
     server_addr_in.sin_family = AF_INET;
     server_addr_in.sin_port = htons(SVR_PORT);
     memcpy(&server_addr_in.sin_addr.S_un.S_addr, ent->h_addr_list[0], 4);
 
-    // server inialialization
+    // Server initialization.
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why

File: modules/rostests/apitests/ws2_32/WSAAsync.c [hunk 7]
```diff
@@ -157,48 +158,48 @@ START_TEST(WSAAsync)
...
     trace("Initializing server and client connections ...\n");
     ok(bind(ServerSocket, (struct sockaddr*)&server_addr_in, sizeof(server_addr_in)) == 0, "ERROR: server bind failed\n");
     ok(ioctlsocket(ServerSocket, FIONBIO, &ulValue) == 0, "ERROR: server ioctlsocket FIONBIO failed\n");
 
-    // client inialialization
+    // Client initialization.
```
Comment (learn-more) [inline modules/rostests/apitests/ws2_32/WSAAsync.c]: Why?

**Review by learn-more — DISMISSED** [review]
It seems to me like you are just touching as many lines as possible,
are you trying to get as many changed lines in the repository?

If it were up to me, i'd reject this issue for that alone.

File: modules/rostests/apitests/ws2_32/WSAAsync.c (hunk 2 above)
Comment (SergeGautherie) [reply modules/rostests/apitests/ws2_32/WSAAsync.c]: `ok(); skip();` reports more explicitly than `skip("ERROR");`, which even looks contradictory...

**Comment by SergeGautherie** [comment]
> are you trying to get as many changed lines in the repository?

No: as PR title reads, I try to improve and fix, a test!

**Comment by binarymaster** [comment]
> **Conflicting files**
> `modules/rostests/apitests/ws2_32/WSAAsync.c`

This PR needs to be updated.

**Comment by SergeGautherie** [comment]
Rebased, after #1132 merge.

**Comment by SergeGautherie** [comment]
> needs rebase

@binarymaster, please update status.

**Comment by SergeGautherie** [comment]
Split function calls and `WSAGetLastError()` reports.

**Comment by SergeGautherie** [comment]
> requested changes

@learn-more, please update status

**Pull request closed without merging**